src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,218 @@
+            const OrgAndRepoDetails = wizardInputs.sourceRepository.repositoryId.split('/');
+            return await this.provisioningServiceClient.createProvisioningConfiguration(provisioningConfiguration, OrgAndRepoDetails[0], OrgAndRepoDetails[1]);
+        } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnableToCreateProvisioningPipeline, error);
+            throw error;
+        }
+    }
+
+    public async getProvisioningPipeline(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+       try {
+        this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+        return await this.provisioningServiceClient.getProvisioningConfiguration(jobId, githubOrg, repository);
+       } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnabletoGetProvisioningPipeline, error);
+            throw error;
+       }
+    }
+
+    public async checkProvisioningPipeline(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration> {
+        try {
+            const provisioningServiceResponse = await this.getProvisioningPipeline(jobId, githubOrg, repository, wizardInputs);
+            if ( provisioningServiceResponse.result.status ===  "Queued" ||  provisioningServiceResponse.result.status == "InProgress") {
+               await sleepForMilliSeconds(this.refreshTime);
+               return await this.checkProvisioningPipeline(jobId, githubOrg, repository, wizardInputs);
+            } else if (provisioningServiceResponse.result.status ===  "Failed") {
+               throw new Error(provisioningServiceResponse.result.message) ;
+            } else {
+                return provisioningServiceResponse;
+            }
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    public async browseQueuedPipeline(): Promise<void> {
+        vscode.window.showInformationMessage(Messages.githubWorkflowSetupSuccessfully, Messages.browseWorkflow)
+            .then((action: string) => {
+                if (action && action.toLowerCase() === Messages.browseWorkflow.toLowerCase()) {
+                    telemetryHelper.setTelemetry(TelemetryKeys.BrowsePipelineClicked, 'true');
+                    vscode.env.openExternal(vscode.Uri.parse(this.queuedPipelineUrl));
+                }
+            });
+    }
+
+    public async postSteps(provisioningConfiguration: ProvisioningConfiguration, draftPipelineConfiguration: DraftPipelineConfiguration, inputs: WizardInputs): Promise<void> {
+        await this.getFileToCommit(draftPipelineConfiguration);
+        await this.showPipelineFiles();
+        const displayMessage = (this.filesToCommit.length > 1) ?  Messages.modifyAndCommitMultipleFiles : Messages.modifyAndCommitFile;
```

Comment:
this should be done by caller. See how it was being done currently. it also should set proper currentstep

Reply:
Currently also we are doing this in configurer https://github.com/microsoft/vscode-deploy-azure/blob/6df6bc5b56db47f932259832c00fbcc8d38e262b/src/configure/configurers/localGithubWorkflowConfigurer.ts#L201

Reply:
ok